### ISSUE TITLE
[UI] multiple GET errors

### ISSUE THREAD

**Issue by hadim** [argoproj/argo#4319]
See https://github.com/argoproj/argo/issues/4297 for context. I open a new issue because the previous one was in fact two issues. I also originally thought the issue was related to HTTP2 but it appears argo does not support HTTP2 so it must be something else.

Argo version tested: 2.11.3, 2.11.4, and 2.11.5.

### Error description

When navigating on `/workflows/argo` or an event page such as `/workflows/argo/awesome-rhino?tab=events&nodeId=null`, after about a minute, the main HTML div of the page is replaced byt the following error:

![Screenshot from 2020-10-18 08-30-02](https://user-images.githubusercontent.com/528003/96368196-9d436f00-1120-11eb-9da8-0da88f59daf7.png)

The browser's console shows:

```bash
argo:1 GET https://xxxxxxx.com/api/v1/workflow-events/argo?listOptions.resourceVersion=57501459&fields=result.object.metadata.name,result.object.metadata.namespace,result.object.metadata.resourceVersion,result.object.metadata.uid,result.object.status.finishedAt,result.object.status.phase,result.object.status.startedAt,result.type,result.object.metadata.labels,result.object.spec.suspend 502
argo:1 EventSource failed loading: GET "https://xxxxxxx.com/api/v1/workflow-events/argo?listOptions.resourceVersion=57501459&fields=result.object.metadata.name,result.object.metadata.namespace,result.object.metadata.resourceVersion,result.object.metadata.uid,result.object.status.finishedAt,result.object.status.phase,result.object.status.startedAt,result.type,result.object.metadata.labels,result.object.spec.suspend".
```

(seen on `/workflows/argo`)

On an event page the error looks like:

```bash
awesome-rhino:1 GET https://xxxx.com/api/v1/workflow-events/argo?listOptions.fieldSelector=metadata.name=awesome-rhino net::ERR_HTTP2_PROTOCOL_ERROR 200
awesome-rhino:1 GET https://xxxx.com/api/v1/workflow-events/argo?listOptions.fieldSelector=metadata.name=awesome-rhino net::ERR_HTTP2_PROTOCOL_ERROR 200
```

I don't see any error on the argo-server logs beside some `info` logs similar to `msg="finished streaming call with code OK"` and ` msg="finished unary call with code OK"`.

### Setup used

Argo is installed on GKE (I tried both `1.17.9-gke.1504` and `1.17.9-gke.6300`) using the Ingress setup configuration from the official GKE doc with a static IP address and an autogenerated certificate:

```yaml
apiVersion: v1
kind: Service
metadata:
  name: argo-server-remote
spec:
  type: NodePort
  ports:
    - name: web
      port: 2746
      targetPort: 2746
      protocol: TCP
  selector:
    app: argo-server-remote
---
apiVersion: networking.gke.io/v1beta2
kind: ManagedCertificate
metadata:
  name: argo-remote-certificate
spec:
  domains:
    - xxxxx.xxxxx.com
---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: argo-remote-ingress
  annotations:
    kubernetes.io/ingress.global-static-ip-name: xxxx-test-ip
    networking.gke.io/managed-certificates: argo-remote-certificate
spec:
  backend:
    serviceName: argo-server-remote
    servicePort: 2746
```

- The argo server is run with the following args: `args: [server, --auth-mode, client, --namespaced]`
- The error appears when using both HTTP and HTTPS.
- The error appears using the cluster's URL as well the cluster's IP.
- It's also probably worth noting that the **error DOES ALSO APPEAR** when accessing argo through a kubectl proxy on the argo `Service` (so the error itself is unlikely related to the `Ingress` setup.

---

I tried to be the most descriptive possible, let me know if you need more infos.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by hadim** [comment]
I forgot to mention two things:

- the error seems to appear kind of randomly when streaming or watching the detail of a running workflow (this one is harder to reproduce).
- this issue is not critical since the workflows can still be submitted, logs can be streamed and all the regular operations work as excepted. But it makes the global UX experience very painful when working with the UI for a long time.

**Comment by alexec** [comment]
I think we fixed this in v2.12, but did not back port due to it being conflicty . Would you be interested in submitting a PR to back port that change to release-2.11

**Comment by hadim** [comment]
Can you point me to the related PR? I am happy to have a quick look but can't commit to anything due to limited bandwidth.

**Comment by alexec** [comment]
```
> feat(ui): Improve error recovery. Close #4087 (#4094)
```

Also, you could try running `argoproj/argocli:latest` to see if that fixes it.

**Comment by eestewart** [comment]
I didn't open the issue, but running `argoproj/argocli:latest` fixed this problem for me.

**Comment by hadim** [comment]
@eestewart thanks for confirming it works on `master`. I probably won't fix the bug in 2.11 and will wait for 2.12 instead. 

I close since this is fixed on `master`.
